### PR TITLE
Move environment variable support from `nautobot_config.py.j2` to `settings.py`, remove `nautobot_config.append.py`, etc.

### PR DESCRIPTION
**Pull request by glennmatthews**
# Closes: #2578 
# What's Changed

- Various environment variables are now supported in `settings.py` itself, rather than having fixed values in `settings.py` and only checking for environment variables in the config file generated from `nautobot_config.py.j2` (`nautobot-server init`)
- Fixed a few environment variables that *were* supported in `settings.py` already but used different variables than our documentation or `nautobot_config.py.j2` used (`SECRET_KEY` -> `NAUTOBOT_SECRET_KEY`, `NAUTOBOT_DATABASE` -> `NAUTOBOT_DB_NAME`, `NAUTOBOT_USER` -> `NAUTOBOT_DB_USER`, `NAUTOBOT_PASSWORD` -> `NAUTOBOT_DB_PASSWORD`)
    - This fix is a potential breaking change if and only if:
       1. A user isn't using a config file generated from `nautobot_config.py.j2` (i.e., via `nautobot-server init`) OR they have removed the relevant lines from their config file, reverting to Nautobot's `settings.py` behavior
       2. AND they are relying on the undocumented environment variables previously coded into `settings.py`
    - I think this is an acceptable risk.
- Removed code in `development/nautobot_config.py` that was copy-pasted from the template and is now redundant with `settings.py`.
- Removed code in `nautobot/core/tests/nautobot_config.py` that was copy-pasted from the template and is now redundant with `settings.py`.
- Moved the `LOGGING` configuration already included in our Docker images (via `nautobot_config.append.py` as a Dockerfile step) into `settings.py` to replace the generic Django defaults and ensure that the default logging behavior is the same for virtualenv versus Docker deployments
     - This too is *potentially* breaking if someone is relying on the default (empty) Django `LOGGING = []`, but I think that also is an acceptable risk
     - This allows us to remove `nautobot_config.append.py` and just use `nautobot-server init` to generate the `nautobot_config.py` for our Docker images.
- In our `nautobot_config.py.j2` template for `nautobot-server init`:
   - Comment out all configs that are now identical to `settings.py`, providing documentation as to their current defaults as well as how to customize them, but not actually overriding `settings.py` by default.
   - Regroup configs into "Required Settings" (as defined by our documentation), "Optional Django core settings", and "Optional Settings for Nautobot and its dependencies".
   - Add commented-out and briefly described examples of all configs that we describe in our documentation, making it easier to customize them as desired.
- Update documentation.

I've done my best to provide thorough notes in the changelog about potential impacts, but would absolutely appreciate feedback.

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- n/a Attached Screenshots, Payload Example
- n/a Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- n/a Example Plugin Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Review by jathanism — APPROVED** [review]
This makes me happy. Only thing I wonder is if we should just make the `nautobot_config.py.j2` file used for the `init-`generated config as minimal as possible? And only leave in the most common things that *may* need to be overloaded or defined?

**Comment by glennmatthews** [comment]
Trying to guess which settings a given deployment will or won't want to overload feels like an impossible task to me. I *think* the approach here (override little or nothing by default, but include copious comments about things you *might* want to override) feels like a good compromise to me, but if we'd rather just strip it down to nearly empty instead I can get onboard with that.

File: nautobot/core/settings.py [hunk 1]
```diff
@@ -597,16 +638,16 @@
     "virtualization.*": {"ops": "all"},
 }
 CACHEOPS_DEGRADE_ON_FAILURE = True
-CACHEOPS_ENABLED = True
-CACHEOPS_REDIS = "redis://localhost:6379/1"
-CACHEOPS_DEFAULTS = {"timeout": 900}
+CACHEOPS_ENABLED = is_truthy(os.getenv("NAUTOBOT_CACHEOPS_ENABLED", "True"))
```
[the remark this answers is not in the document]
Comment (jathanism) [reply nautobot/core/settings.py:641]: Did you already handle this or should we backlog it?

[the remark this answers is not in the document]
Comment (glennmatthews) [reply nautobot/core/settings.py:641]: Haven't yet; wasn't sure when our next merge up into `next` will be. If we're okay with doing a merge at present I can handle it as time permits.

**Pull request merged**